ironfish/src/network/peerNetwork.ts

```diff
@@ -993,6 +1021,69 @@ export class PeerNetwork {
     return await this.chain.getHeaderAtSequence(start)
   }
 
+  private async onGetBlockHeadersRequest(
+    request: IncomingPeerMessage<GetBlockHeadersRequest>,
+  ): Promise<GetBlockHeadersResponse> {
+    const peer = this.peerManager.getPeerOrThrow(request.peerIdentity)
+    const rpcId = request.message.rpcId
+
+    if (request.message.limit === 0) {
+      peer.punish(
+        BAN_SCORE.LOW,
+        `Peer sent GetBlockHeaders with limit of ${request.message.limit}`,
+      )
+      return new GetBlockHeadersResponse([], rpcId)
+    }
+
+    if (request.message.limit > MAX_REQUESTED_BLOCKS) {
```

Comment:
This could be done in a separate PR, but I think we could consider creating a separate MAX_REQUESTED_HEADERS value and increasing it, since block headers are much smaller than blocks. I'd probably take a look around bitcoin and/or eth and see what they consider reasonable values.